call-home/src/bin/callhome/main.rs

```diff
@@ -193,14 +194,10 @@ async fn generate_report(
         }
     };
 
-    let volumes = http_client.volumes_api().get_volumes(0, None, None).await;
-    let volumes = match volumes {
-        Ok(volumes) => Some(volumes.into_body()),
-        Err(err) => {
-            error!("{:?}", err);
-            None
-        }
-    };
+    let volumes = list_all_volumes(&http_client)
+        .await
+        .map_err(|error| error!("{error:?}"))
```

Comment:
This bit was also a bit confusing from the logs, maybe include what the error was, ex:
```suggestion
        .map_err(|error| error!("Failed to list all volumes: {error:?}"))
```